client/dashboard/me/notifications-extras/extras-toggle-card/test/index.test.tsx

```diff
@@ -0,0 +1,239 @@
+		expect(
+			screen.getByText( 'Opportunities to participate in WordPress.com research and surveys.' )
+		).toBeVisible();
+		expect(
+			screen.getByText( 'Information on WordPress.com courses and events (online and in-person).' )
+		).toBeVisible();
+	} );
+
+	it( 'shows "Subscribe to all" when no options are enabled', () => {
+		renderExtrasToggleCard();
+
+		expect( screen.getByRole( 'checkbox', { name: 'Subscribe to all' } ) ).toBeVisible();
+		expect( screen.queryByText( 'Unsubscribe from all' ) ).not.toBeInTheDocument();
+	} );
+
+	it( 'shows "Unsubscribe from all" when some options are enabled', () => {
+		const extraSettings = {
+			...defaultExtraSettings,
+			marketing: true,
+			research: true,
+		};
+
+		renderExtrasToggleCard( { extraSettings } );
+
+		expect( screen.getByRole( 'checkbox', { name: 'Unsubscribe from all' } ) ).toBeVisible();
+		expect( screen.queryByText( 'Subscribe to all' ) ).not.toBeInTheDocument();
+	} );
+
+	it( 'reflects current settings in toggle states', () => {
+		const extraSettings = {
+			...defaultExtraSettings,
+			marketing: true,
+			community: true,
+			reports: true,
+		};
+
+		renderExtrasToggleCard( { extraSettings } );
+
+		expect( screen.getByLabelText( 'Suggestions' ) ).toBeChecked();
+		expect( screen.getByLabelText( 'Research' ) ).not.toBeChecked();
+		expect( screen.getByLabelText( 'Community' ) ).toBeChecked();
+		expect( screen.getByLabelText( 'Promotions' ) ).not.toBeChecked();
+		expect( screen.getByLabelText( 'Reports' ) ).toBeChecked();
+	} );
+
+	it( 'calls onMutate with only changed item when individual toggle is changed', async () => {
+		renderExtrasToggleCard();
+
+		const marketingToggle = screen.getByLabelText( 'Suggestions' );
+		await userEvent.click( marketingToggle );
+
+		expect( mockOnMutate ).toHaveBeenCalledWith( { marketing: true } );
+	} );
+
+	it( 'calls onMutate when "Subscribe to all" is clicked', async () => {
+		renderExtrasToggleCard();
+
+		const subscribeAllToggle = screen.getByLabelText( 'Subscribe to all' );
+		await userEvent.click( subscribeAllToggle );
+
+		expect( mockOnMutate ).toHaveBeenCalledWith( {
+			marketing: true,
+			research: true,
+			community: true,
+			promotion: true,
+			news: true,
+			digest: true,
+			reports: true,
+			news_developer: true,
+			scheduled_updates: true,
+		} );
+	} );
+
+	it( 'calls onMutate with only changed values when "Unsubscribe from all" is clicked', async () => {
+		const extraSettings = {
+			...defaultExtraSettings,
+			marketing: true,
+			research: true,
+			community: true,
+		};
+
+		renderExtrasToggleCard( { extraSettings } );
+
+		const unsubscribeAllToggle = screen.getByLabelText( 'Unsubscribe from all' );
+		await userEvent.click( unsubscribeAllToggle );
+
+		expect( mockOnMutate ).toHaveBeenCalledWith( {
+			marketing: false,
+			research: false,
+			community: false,
+		} );
+	} );
+
+	it( 'disables all toggles when isSaving is true', () => {
+		renderExtrasToggleCard( { isSaving: true } );
```

Comment:
Minor Note: Should it be disabled instead of `isSaving` since the inner component really doesn't have the concept of saving?  Another option is `isBusy` to be aligned with the Gutenberg implementation.